fence/agents/compute/fence_compute.py

```diff
@@ -229,24 +229,20 @@ def fix_domain(options):
 				# Compute nodes are named as FQDN, strip off the hostname
 				calculated = service.host.replace(shorthost+".", "")
 
-			domains[calculated] = shorthost
-
 			if calculated == last_domain:
 				# Avoid complaining for each compute node with the same name
 				# One hopes they don't appear interleaved as A.com B.com A.com B.com
 				logging.debug("Calculated the same domain from: %s" % service.host)
+				continue
 
-			elif "--domain" in options and options["--domain"] == calculated:
-				# Supplied domain name is valid 
-				return
```

Comment:
But the warning about node with a different domain is important. And really, a loop over 1000 nodes will take less that 0.05s. That's much less than a call to the nova API.